src/classes/CRLP_RollupUI_SVC.cls

```diff
@@ -431,4 +480,43 @@ public with sharing class CRLP_RollupUI_SVC {
         return labels;
     }
 
+    /*******************************************************************************************************
+    * @description Finds writable, custom, unused target fields of specific types to provide field lists for Rollup definitions.
+    *              Includes an exception for a set summary field on the rollup detail.
+    * @param objectName the name of the object to look up
+    * @param selectedField the name of a summary field already in use
+    * @return List<Map<String, String>> an object with field api names, labels and data type.
+    */
+    public static List<Map<String, String>> getTargetFieldsWithType(String objectName, String selectedField) {
+
+        List<Map<String, String>> fieldObjectList = new List<Map<String, String>>();
+
+        Map<String, Schema.DescribeFieldResult> objectFields = UTIL_Describe.getAllFieldsDescribe(objectName);
+        Schema.SObjectType sobjType = UTIL_Describe.getSObjectType(objectName);
+        List<Schema.DescribeFieldResult> referencedFields = CRLP_Rollup_SEL.getReferencedFieldsByObject(sobjType);
+        Set<Schema.DescribeFieldResult> referencedFieldSet = new Set<Schema.DescribeFieldResult>(referencedFields);
+
+        for (Schema.DescribeFieldResult dfr : objectFields.values()) {
+            //valid types are taken from the DisplayType Enum and only include values that make sense to roll up
+            String type = dfr.getType().Name();
+            Set<String> validTypeSet = new Set<String>{'DATE', 'DATETIME', 'DOUBLE', 'ID', 'MULTIPICKLIST', 'PICKLIST', 'STRING', 'TEXTAREA'};
+            Boolean isValidType = validTypeSet.contains(type);
+
+            //check that summary field isn't being used on another rollup
+            Boolean isRolledUp = referencedFieldSet.contains(dfr);
+
+            if((dfr.isUpdateable() && dfr.isCustom() && !dfr.name.contains('system') && isValidType && !isRolledUp)
```

Comment:
since I'm checking the name instead of the label, do I need to check for the _? Or should I be checking the label instead?

Reply:
It would have to look for `__system` in the api name for it to work. The label can't be guaranteed upon to have "System" in it if the custom has renamed or translated the field.